atom/browser/net/url_request_stream_job.cc

```diff
@@ -28,6 +28,13 @@ URLRequestStreamJob::URLRequestStreamJob(net::URLRequest* request,
       response_headers_(nullptr),
       weak_factory_(this) {}
 
+URLRequestStreamJob::~URLRequestStreamJob() {
+  if (subscriber_) {
+    content::BrowserThread::DeleteSoon(content::BrowserThread::UI, FROM_HERE,
```

Comment:
Didn't knew this function existed :+1: 

Anyway, the only reason I added `SafePtr` was to encapsulate "deletion in the UI thread" in case other classes that are deleted outside UI thread needed to use `EventSubscriber` and didn't want to repeat this code in their destructors.